test/Test/Toml/Parsing/Unit.hs

```diff
@@ -20,28 +22,38 @@ spec_Parser = do
       failOn p given            = parse p "" `shouldFailOn` given
       parseXSatisfies p given f = parse p "" given `parseSatisfies` f
 
-      parseArray   = parseX arrayP
-      parseBool    = parseX boolP
-      parseDouble  = parseX doubleP
-      parseInteger = parseX integerP
-      parseKey     = parseX keyP
-      parseKeyVal  = parseX keyValP
-      parseText    = parseX textP
-      parseTable   = parseX tableHeaderP
-      parseToml    = parseX tomlP
+      parseArray    = parseX arrayP
+      parseBool     = parseX boolP
+      parseDateTime = parseX dateTimeP
+      parseDouble   = parseX doubleP
+      parseInteger  = parseX integerP
+      parseKey      = parseX keyP
+      parseKeyVal   = parseX keyValP
+      parseText     = parseX textP
+      parseTable    = parseX tableHeaderP
+      parseToml     = parseX tomlP
 
-      arrayFailOn  = failOn arrayP
-      boolFailOn   = failOn boolP
-      doubleFailOn = failOn doubleP
-      keyValFailOn = failOn keyValP
-      textFailOn   = failOn textP
+      arrayFailOn    = failOn arrayP
+      boolFailOn     = failOn boolP
+      dateTimeFailOn = failOn dateTimeP
+      doubleFailOn   = failOn doubleP
+      keyValFailOn   = failOn keyValP
+      textFailOn     = failOn textP
 
       doubleSatisfies = parseXSatisfies doubleP
 
       quoteWith q t = q <> t <> q
       squote        = quoteWith "'"
       dquote        = quoteWith "\""
 
+      makeDay year month day            = Day $ fromGregorian year month day
```

Comment:
@ChShersh I have tried here to do: `makeDay = Day . fromGregorian`, but it doesn't work because `fromGregorian` takes accepts 3 arguments.

Is there a workaround that allows doing `makeDay = Day . fromGregorian` ?

Reply:
@ghallak There exist trick for variadic composition operator. We implemented it in `universum`:

* https://github.com/serokell/universum/blob/master/src/Universum/VarArg.hs

But with usual composition operator you can't do that. So `makeDay year month day = Day $ fromGregorian year month day`